nav2_util/include/nav2_util/validate_messages.hpp

```diff
@@ -0,0 +1,106 @@
+//  Check recieved message is safe or not for the nav2-system
+//  For each msg-type known in nav2, we could check it as following:
+//  if(!validateMsg()) RCLCPP_ERROR(,"malformed msg. Rejecting.")
+//
+//  Workflow of validateMsg():
+//     if here's a sub-msg-type in the recieved msg,
+//        the content of sub-msg would be checked as sub-msg-type
+//     then, check the whole recieved msg.
+//
+//  Following conditions are involved in check:
+//     1> Value Check: to avoid damaged value like like `nan`, `INF`
+//     2> Logic Check: to avoid value with bad logic, like the size of `map` should be equal to `height*width`
+//     3> Any other needed condition could be joint here in future
+
+#ifndef validate_message_HPP
+#define validate_message_HPP
+
+
+#include "nav_msgs/msg/occupancy_grid.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+
+
+namespace nav2_util
+{
+
+const int NSEC_PER_SEC = 1e9;  // 1 second = 1e9 nanosecond
+
+bool validateMsg(const builtin_interfaces::msg::Time_<std::allocator<void>> & msg)
+{
+  // check item
+  if (msg.sec < 0) {return false;}                                      // invalid second-stamp
+  if (msg.nanosec < 0 || msg.nanosec >= NSEC_PER_SEC) {return false;}   // invalid nanosec-stamp
+  return true;
+}
+
+bool validateMsg(const std_msgs::msg::Header_<std::allocator<void>> & msg)
+{
+  //  check sub-type
+  if (!validateMsg(msg.stamp)) {return false;}
+  // msg.frame_id :  @todo any check for it ?
+  return true;
+}
+
+bool validateMsg(const geometry_msgs::msg::Point & msg)
+{
+  // @todo any check for following item ?
+  msg.x;
+  msg.y;
+  msg.z;
+  return true;
+}
+bool validateMsg(const geometry_msgs::msg::Quaternion & msg)
+{
+  // @todo any check for following item ?
+  msg.x;
```

Comment:
NaN or Inf for any values

It could also be checked that its normalized: https://math.stackexchange.com/questions/1703466/normalizing-a-quaternion

If its not normalized within some `eta` value, then reject

Reply:
I'm not entirely sure if a precision error tolerance like epsilon = 1e-6 is appropriate; I'll test it locally later.